modules/pam_motd/pam_motd.c

```diff
@@ -166,9 +166,24 @@ static int compare_strings(const void *a, const void *b)
     }
 }
 
+static const char* motddir;
```

Comment:
Static non-const variable is problematic for those of pam modules that are intended to be MT-safe.

Reply:
yes - is motd supposed to be mt-safe?  it's using scandir(), which in turn uses readdir().  
Do you have a suggestion of an alternative solution?

Reply:
I think this might solve it - comments?
```
static const char* motddir( char *d )
{
    static const char* dir;

    if ( d ) dir=d;

    return dir;
}
```

Reply:
> yes - is motd supposed to be mt-safe? it's using scandir(), which in turn uses readdir(). Do you have a suggestion of an alternative solution?

scandir(3) manual page claims that scandir() is MT-safe.
I suppose `__thread` qualifier can be used if available.